### PR TITLE
feat: [54] Pay cycle settings + Buffer Until Next Pay dashboard widget

### LINKED ISSUE #54: Pay cycle detection + Buffer Until Next Pay feature
## Description

Add the second core dashboard number: "Buffer Until Next Pay" — showing whether the user is above or below what they need before their next pay cycle. This answers the "Can I afford this?" question with context.

## Why

"Available to Spend" alone doesn't account for upcoming committed expenses. A user might have $2,000 available but $1,800 in bills due before payday. The buffer number closes that gap: "+$200 above what you need" or "-$300 below what you need".

### Schema Changes

Migration: `add_pay_cycle_settings_to_users_table`
- `pay_amount` bigInteger nullable — expected pay per cycle (cents)
- `pay_frequency` string nullable — weekly / fortnightly / monthly
- `next_pay_date` date nullable
- `committed_per_cycle` bigInteger nullable — "I need $X per cycle for essentials" (cents)

### Tasks

- [ ] Create migration for pay cycle columns on users table
- [ ] Add settings UI: "How much do you get paid?", "How often?", "When's your next payday?", "How much do you need per cycle?"
- [ ] Buffer calculation: `Available to Spend - committed_per_cycle`
- [ ] Dashboard display: "+$430 above" (green) or "-$200 below" (red) what you need
- [ ] Write Pest tests for buffer calculation and component rendering

### Settings UI

Simple form on a new settings section or page:
- Pay amount (dollar input → stored as cents)
- Frequency dropdown (weekly / fortnightly / monthly)
- Next pay date (date picker)
- Committed amount per cycle (dollar input → stored as cents)

## Phase B: Auto-Detection (Future Issue)

- Analyze primary account for largest recurring credit deposit
- Detect frequency by date pattern analysis
- Auto-populate pay settings on first setup
- Detect recurring debits and auto-suggest committed amount
- _This phase will be a separate issue_

## Acceptance Criteria

- [ ] User can configure pay cycle in settings
- [ ] Dashboard shows buffer number alongside "Available to Spend"
- [ ] Buffer is green when positive, red when negative
- [ ] Settings persist across sessions
- [ ] Pest tests cover calculation logic and edge cases (no settings configured, zero buffer, negative buffer)
- [ ] `op test` passes

## Files to Create/Modify

- New migration: `add_pay_cycle_settings_to_users_table`
- `app/Models/User.php` — new casts for pay cycle fields
- New or updated Livewire component for buffer display
- New settings view/component for pay cycle configuration
- `resources/views/dashboard.blade.php`
- New test files

## Depends On

- #53 (Available to Spend widget)

### PR DESCRIPTION
**Pull request by robwilde**
## Summary

- Adds pay cycle configuration (pay amount, frequency, next pay date, committed expenses) to user settings
- Displays "Buffer Until Next Pay" on the dashboard below "Available to Spend" — green when above committed, red when below, with a "Set up pay cycle →" CTA when not configured
- Creates `PayFrequency` enum (`Weekly`, `Fortnightly`, `Monthly`)
- Adds `bufferUntilNextPay()` and `hasPayCycleConfigured()` methods to User model
- Fixes pre-existing PHPStan issue with `when()` helper in settings routes

## Files Changed

**New files:**
- `app/Enums/PayFrequency.php`
- `database/migrations/*_add_pay_cycle_settings_to_users_table.php`
- `resources/views/pages/settings/⚡pay-cycle.blade.php`
- `tests/Unit/Enums/PayFrequencyTest.php`
- `tests/Feature/Settings/PayCycleSettingsTest.php`
- `tests/Browser/Livewire/AccountOverviewBrowserTest.php`
- `tests/Browser/Settings/PayCycleSettingsBrowserTest.php`

**Modified files:**
- `app/Models/User.php` — pay cycle fields, casts, buffer methods
- `database/factories/UserFactory.php` — `withPayCycle()` state
- `routes/settings.php` — pay-cycle route + `when()` closure fix
- `resources/views/pages/settings/layout.blade.php` — nav item
- `app/Livewire/AccountOverview.php` — buffer calculation
- `resources/views/livewire/account-overview.blade.php` — buffer display
- `tests/Feature/Models/UserTest.php` — buffer tests
- `tests/Feature/Livewire/AccountOverviewTest.php` — buffer tests

## Test plan

- [ ] `op test` — 371 tests pass
- [ ] `op test.browser` — 19 browser tests pass
- [ ] `op analyse` — PHPStan clean
- [ ] `op lint.dirty` — Pint clean
- [ ] Visit `/settings/pay-cycle`, fill form, save, verify values persist on reload
- [ ] Visit `/dashboard`, confirm buffer displays below "Available to Spend"
- [ ] Verify green/red/neutral states for positive/negative/zero buffer
- [ ] Verify "Set up pay cycle →" CTA appears when no pay cycle configured

Closes #54

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -4,16 +4,25 @@
...
 use Laravel\Fortify\TwoFactorAuthenticatable;
 
-/** @property CarbonImmutable|null $last_synced_at */
+/**
+ * @property CarbonImmutable|null $last_synced_at
+ * @property int|null $pay_amount
+ * @property PayFrequency|null $pay_frequency
+ * @property Carbon|null $next_pay_date
+ * @property int|null $committed_per_cycle
+ */
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Models/User.php:24]: Fixed — updated PHPDoc to `CarbonImmutable|null` and removed the unused `Illuminate\Support\Carbon` import.

File: resources/views/pages/settings/layout.blade.php [hunk 2]
```diff
@@ -1,12 +1,13 @@
 <div class="flex items-start max-md:flex-col">
-    <div class="me-10 w-full pb-4 md:w-[220px]">
+    <div class="me-10 w-full pb-4 md:w-55">
```
[the remark this answers is not in the document]
Comment (robwilde) [reply resources/views/pages/settings/layout.blade.php]: Fixed — changed to `md:w-56` (standard Tailwind v4 spacing scale value).

File: app/Livewire/AccountOverview.php [hunk 3]
```diff
@@ -36,11 +36,15 @@ public function render(): View
             ->filter(fn (Account $a) => $a->type->isSpendable())
             ->sum(fn (Account $a) => $a->availableBalance());
 
+        $user = auth()->user();
+        $buffer = $user->bufferUntilNextPay($availableToSpend);
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Livewire/AccountOverview.php]: Fixed — moved `$user = auth()->user()` to top of `render()` and reused it for the accounts query.
